### DISCUSSION TITLE
Allow annotations for problem matchers to suggest fixes

### DISCUSSION THREAD

**Discussion by nschonni** [community/community#16742] [Other Feature Feedback, Questions, & Ideas]
Some linting tools allow "fix" suggestions (ESLint, Markdownlint, etc...), so it would be useful for the built-in annotations to allow suggested edits in addition to the current warning/error/notice ones currently available.
This would allow the simple whitespace/formatting issues to be added to a batch by a PR author. Maybe this would require a new hook instead of "registering a problem matcher", for "registering a problem fixer"

**Comment by ZedThree** [comment]
+1 Yes please! I have an Action that comments with the fixits, which enables users to directly apply them to the PR. There are various issues with using comments, including running from forks, out of date comments, spamming the PR, etc.
Annotations would solve at least some of these issues, but the limited number (10 per step) and the inability to add suggestions/fixits means they're just not as useful.
